task/bq2bq/executor/bumblebee/handler.py

```diff
@@ -1,27 +1,51 @@
 import signal
 import sys
 from bumblebee.log import get_logger
+import threading
+import time
 
 logger = get_logger(__name__)
 
 class BigqueryJobHandler:
     def __init__(self) -> None:
         self._sum_slot_millis = 0
         self._sum_total_bytes_processed = 0
+        self.client = None
+        self.jobs = []
+        self.is_done = False
+        self.is_cancel = False
+        self._thread = threading.Thread(target=self._handle_state)
```

Comment:
self._thread might not be required. in the handle_sigterm we can just kill all jobs !